src/main/java/io/synadia/flink/v0/sink/writer/NatsJetStreamSinkWriter.java

```diff
@@ -30,13 +27,9 @@ public NatsJetStreamSinkWriter(String sinkId,
     @Override
     public void write(InputT element, Context context) throws IOException, InterruptedException {
         byte[] payload = payloadSerializer.getBytes(element);
+
         for (String subject : subjects) {
-            try {
-                ctx.js.publish(subject, null, payload);
-            }
-            catch (JetStreamApiException e) {
-                throw new FlinkRuntimeException(e);
-            }
```

Comment:
No. Maybe have a sink constructor that indicates sync or async publish but that's not my main objections. The publish has to succeed, and with the new async impl, you still have to check that the future completed and that's not done here